pkg/server/node.go

```diff
@@ -250,8 +250,19 @@ func (s *nodeServer) attest(
 	attestedData *common.AttestedData, attestedBefore bool) (
 	response *nodeattestor.AttestResponse, err error) {
 
-	// TODO: Pick the right node attestor [#222]
-	nodeAttestor := s.catalog.NodeAttestors()[0]
+	// Pick the right node attestor
+	var nodeAttestor nodeattestor.NodeAttestor
+	for _, a := range s.catalog.NodeAttestors() {
+		mp := s.catalog.Find(a)
+		if mp.Config.PluginName == attestedData.Type {
```

Comment:
Does this imply attestedData Type need to match the pluginName

Reply:
Yes... `Type` is probably a bad choice for name here, but changing the name of the field would be cross-cutting. Opened https://github.com/spiffe/spire/issues/289 to track, feel free to add comments there